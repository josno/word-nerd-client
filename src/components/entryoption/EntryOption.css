a {
	text-decoration: none;
}

.demo-information {
	font-size: 1rem;
	font-style: italic;
}

.entry-form {
	margin-left: auto;
	margin-right: auto;
	height: fit-content;
	width: 80vw;
}

.entry-option-container {
	background-color: #18c0f0;
	padding: 20px;
}

.error-message {
	color: red;
	font-weight: bold;
	font-style: italic;
}

.form-input::placeholder {
	color: white;
}

.form-tab,
.input-submit-button {
	border-radius: 15px;
	display: inline-block;
	padding: 10px;
	border: 0.5px solid black;
	margin: 10px;
	font-size: 1.5em;
	background-color: #ff206e;
	box-shadow: 5px 5px #ff206e;
	color: white;
}

.form-tab:hover,
.input-submit-button:hover {
	cursor: pointer;
	background-color: #ffffff;
	color: #ff206e;
	outline: none;
}

.formtab:focus,
.formtab:visited {
	outline: none;
}

.form-tab-list-item {
	list-style: none;
	display: inline-block;
}

.form-input {
	margin-left: auto;
	margin-right: auto;
	width: 60vw;
	margin: 10px;
	font-size: 26px;
	padding: 0.4rem;
	background-color: #42cbf2;
	border: none;
}

ul {
	padding: 0px;
}

@media (min-width: 600px) {
	.entry-form {
		width: 50vw;
	}

	.form-input {
		width: 30vw;
	}
}
