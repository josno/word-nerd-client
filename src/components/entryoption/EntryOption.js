import React, { Component } from 'react';
import './EntryOption.css';
import SignUp from '../SignUp/SignUp';
import WordContext from '../../WordContext';
import Login from '../Login/Login';

class EntryOption extends Component {
	static contextType = WordContext;
	constructor(props) {
		super(props);
		this.state = {
			formType: 'login'
		};
		this.toggleToSignUp = this.toggleToSignUp.bind(this);
		this.toggleToLogin = this.toggleToLogin.bind(this);
	}

	toggleToSignUp = () => {
		this.setState({
			formType: 'signUp'
		});
	};

	toggleToLogin = () => {
		this.setState({
			formType: 'login'
		});
	};

	render() {
		const { formType } = this.state;
		return (
			<div className="entry-option-container">
				<div className="entry-form">
					<div className="demo-information">
						<p> Login demo</p>
						<p> user: 'nerd' pass: 'nerdpass'</p>
					</div>
					<ul>
						<li className="form-tab-list-item">
							<button
								className="form-tab sign-up-tab"
								onClick={this.toggleToSignUp}
							>
								Sign Up
							</button>
						</li>

						<li className="form-tab-list-item">
							<button
								className="form-tab login-tab"
								onClick={this.toggleToLogin}
							>
								Login
							</button>
						</li>
					</ul>
					{formType === 'signUp' ? <SignUp /> : <Login />}
				</div>
			</div>
		);
	}
}

export default EntryOption;
