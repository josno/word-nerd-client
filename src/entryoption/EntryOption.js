import React, { Component } from 'react';
import './EntryOption.css';
import SignUp from '../signup/SignUp';
import WordContext from '../WordContext';
import Login from '../login/Login';

class EntryOption extends Component {
	static contextType = WordContext;
	constructor(props) {
		super(props);
		this.state = {
			formType: 'signUp'
		};
		this.toggleToSignUp = this.toggleToSignUp.bind(this);
		this.toggleToLogin = this.toggleToLogin.bind(this);
	}

	toggleToSignUp = () => {
		this.setState({
			formType: 'signup'
		});
	};

	toggleToLogin = () => {
		this.setState({
			formType: 'login'
		});
	};

	render() {
		const { formType } = this.state;
		return (
			<div className="entry-option">
				<div className="form">
					<ul>
						<li className="form-tab-list-item">
							<button
								className="form-tab"
								onClick={this.toggleToSignUp}
							>
								Sign Up
							</button>
						</li>

						<li className="form-tab-list-item">
							<button
								className="form-tab"
								onClick={this.toggleToLogin}
							>
								Login
							</button>
						</li>
					</ul>
					{formType === 'siginUp' ? (
						<SignUp />
					) : formType === 'login' ? (
						<Login />
					) : (
						<SignUp />
					)}
				</div>
			</div>
		);
	}
}

export default EntryOption;
