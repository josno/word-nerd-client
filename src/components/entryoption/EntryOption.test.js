import React from 'react';
// import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';
import renderer from 'react-test-renderer';
import EntryOption from './EntryOption';

describe('<EntryOption/>', () => {
	it('renders as expected', () => {
		const tree = renderer
			.create(
				<BrowserRouter>
					<EntryOption />
				</BrowserRouter>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
