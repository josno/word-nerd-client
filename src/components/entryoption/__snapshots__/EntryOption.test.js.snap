// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EntryOption/> renders as expected 1`] = `
<div
  className="entry-option"
>
  <div
    className="form"
  >
    <ul>
      <li
        className="form-tab-list-item"
      >
        <button
          className="form-tab"
          onClick={[Function]}
        >
          Sign Up
        </button>
      </li>
      <li
        className="form-tab-list-item"
      >
        <button
          className="form-tab"
          onClick={[Function]}
        >
          Login
        </button>
      </li>
    </ul>
    <div
      className="login"
    >
      <form
        onSubmit={[Function]}
      >
        <div
          className="input-form"
        >
          <input
            className="form-input"
            htmlFor="username"
            name="username"
            placeholder="username"
            type="text"
          />
          <input
            className="form-input"
            htmlFor="password"
            name="password"
            placeholder="password"
            type="password"
          />
        </div>
        <button
          className="input-submit-button"
          type="submit"
        >
          Login
        </button>
      </form>
      <div
        className="error-message"
      />
    </div>
  </div>
</div>
`;
